### ISSUE TITLE
WARNING: Opening secrets file /etc/smokeping/smokeping_secrets: Permission denied

### ISSUE THREAD

**Issue by tanpengsccd** [dgraziotin/docker-smokeping-slave#1]
```
version: "2.1"
services:
    smokeping-slave:
        image: "dgraziotin/smokeping-slave"
        container_name: smokeping-slave
        hostname: slave.xxx.work
        environment:
            - PUID=1000
            - PGID=1000
            - TZ=Asia/Shanghai
            - ServerName="SlaveSMP"
            - SHARED_SECRET="passwd123"
            - MASTER_URL=http://master.xxx.work:8888/smokeping/smokeping.cgi
        # volumes:
        #   - /data/smokeping/config:/config
        #   - /data/smokeping/data:/data
        #    - /data/smokeping/smokeping:/etc/smokeping
        ports:
            - 8888:80
        restart: unless-stopped
```

```docker-compose up```  get error: WARNING: Opening secrets file /etc/smokeping/smokeping_secrets: Permission denied

**Comment by tanpengsccd** [comment]
``` docker exec -it smokeping-slave bash```
```
root@tccd:/# smokeping --check
ERROR: can't open /usr/bin/../etc/smokeping/config: No such file or directory
root@tccd:/# 
```

**Comment by dgraziotin** [comment]
I can't replicate your first issue with this docker-compose.yml (with real URL, hostname, and secrets):

```
---
version: "2.1"
services:
  smokeping-slave:
    image: dgraziotin/smokeping-slave
    container_name: smokeping-slave
    hostname: my-slave
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=Europe/Berlin
      - SHARED_SECRET=12345
      - MASTER_URL=http://my-master.ext/smokeping/smokeping.cgi
        #    volumes:
        #- ./config:/config
        #- ./data:/data
    restart: unless-stopped
```

```
docker-compose up | grep WARNING
Creating network "smokeping_default" with the default driver
Creating smokeping-slave ... done
```

```
docker exec -it smokeping-slave ls -la /config/smokeping_secrets
-rw-r----- 1 abc abc 25 Jul  6 10:17 /config/smokeping_secrets
```

I do see the error message you have in the second comment, but it does not seem related or even an issue at all. Probably what a slave instance shows. It is otherwise working with me.

Are you using the latest image?

**Comment by Nenq** [comment]
Having a similar issue. Graphs from slave are empty, after doing some reading I made an assumption there are some issues with permissions. https://github.com/oetiker/SmokePing/wiki/FAQ#master-slave-setups

However, I noticed that all files have the owner of 1000 uid, so I am still not sure why graphs are empty from a slave.

`smokeping --check
ERROR: can't open /usr/bin/../etc/smokeping/config: No such file or directory`

edit1: 
I have also tried connecting a slave not from docker container.

```
[root@slave1 bin]# ./smokeping --master-url=http://xxx.xxx.yyy.zzz/smokeping/smokeping.cgi --cache-dir=/etc/smokeping/var/ --shared-secret=/etc/smokeping/etc/smokeping_secrets.dist
WARNING: Opening secrets file /etc/smokeping/smokeping_secrets: Permission denied

ERROR: we did not get config from the master. Maybe we are not configured as a slave for any of the targets on the master ?

[root@slave1 bin]# stat /etc/smokeping/etc/smokeping_secrets.dist
  File: ‘/etc/smokeping/etc/smokeping_secrets.dist’
  Size: 5               Blocks: 8          IO Block: 4096   regular file
Device: 801h/2049d      Inode: 2365554     Links: 1
Access: (0640/-rw-r-----)  Uid: (    0/    root)   Gid: (    0/    root)
Access: 2021-07-12 09:15:52.753546367 -0500
Modify: 2021-07-12 09:10:22.183329868 -0500
Change: 2021-07-12 09:12:07.655675192 -0500
 Birth: -
```

**Comment by dgraziotin** [comment]
I can't reproduce this :-( I have a master (linuxserver.io) running with two slaves (mine). I see the "No such file or directory" error, but everything is working.

<img width="459" alt="Screen Shot 2021-07-13 at 08 41 52" src="https://user-images.githubusercontent.com/153179/125403770-2ccb3500-e3b6-11eb-81a5-ef2a4846f037.png">

In your edit, there's an error message stating that the slave does not receive a configuration at all from the master. Might that be the issue?

Edit: @Nenq long shot and I do not even think that this can be an issue, but I'll try. Does your user have both `UID` and `GID` set to 1000? I notice that on one of my VPS, the default user has got `UID` 1000 but GID 1001. I modified my docker-compose.yml accordingly, so I did not think of it initially.

**Comment by Nenq** [comment]
The error regarding the slave not receiving configuration was just a general error due to the warning regarding the permission.

Nonetheless, I managed to solve it and it was my bad because I misconfigured the smokeping_secrets file on master :(

I managed to solve this by mounting a smokeping_secrets file into a docker container. Previously I mounted it on /config directory instead of /etc where slave expects it to be.

```
    volumes:
      - /opt/smokeping/config:/config
      - /opt/smokeping/data:/data
      - /opt/smokeping/etc/smokeping/smokeping_secrets:/etc/smokeping/smokeping_secrets
```
Hopefully, this post will help someone.

Apologies for wasting your time and thanks for your work!
